pathd/path_pcep_lib.c

```diff
@@ -952,6 +1021,30 @@ void pcep_lib_parse_metric(struct path *path, struct pcep_object_metric *obj)
 	path->first_metric = metric;
 }
 
+void pcep_lib_parse_endpoints_ipv4(struct path *path,
+				   struct pcep_object_endpoints_ipv4 *obj)
+{
+	SET_IPADDR_V4(&path->pcc_addr);
+	path->pcc_addr.ipaddr_v4 = obj->src_ipv4;
+	SET_IPADDR_V4(&path->nbkey.endpoint);
+	path->nbkey.endpoint.ipaddr_v4 = obj->dst_ipv4;
+}
+
+void pcep_lib_parse_endpoints_ipv6(struct path *path,
+				   struct pcep_object_endpoints_ipv6 *obj)
+{
+	SET_IPADDR_V6(&path->pcc_addr);
+	path->pcc_addr.ipaddr_v6 = obj->src_ipv6;
+	SET_IPADDR_V6(&path->nbkey.endpoint);
+	path->nbkey.endpoint.ipaddr_v6 = obj->dst_ipv6;
+}
+
+void pcep_lib_parse_vendor_info(struct path *path,
+				struct pcep_object_vendor_info *obj)
+{
+	path->nbkey.color = obj->enterprise_color;
+}
+
```

Comment:
Here you assume that vendor info convey only the color. However, if you use another commercial tool e.g. Juniper Northstar, you have great chance to get a Vendor Info that corresponds to another subobject. Thus, it will be safe to parse and verify the Vendor Object and only store color if it is corresponds to what you expected i.e. what you supported. Otherwise, simply ignore this vendor object

Reply:
I'll conditioned that to cisco vendor which is the supported one right now.